src/Tabs.jsx

```diff
@@ -0,0 +1,180 @@
+import React from 'react';
+import cx from 'classnames';
+
+/**
+ * @module TabsListTab
+ *
+ * Direct child of `TabsList`.
+ * Renders individual tab control.
+ * Accepts a `selected` prop.
+ */
+export const TabsListTab = React.createClass({
+	propTypes: {
+		selected: React.PropTypes.bool,
+		onClick: React.PropTypes.func,
+	},
+	defaultProps: {
+		selected: false
+	},
+	render() {
+		const {
+			children,
+			className,
+			selected,
+			onClick,
+			tabsRef,
+			tabsIndex,
+			...other
+		} = this.props;
+
+		const classNames = cx(
+			'tabs-tab',
+			{
+				'tabs-tab--selected': selected
+			},
+			className
+		);
+
+		return (
+			<li
+				role='tab'
+				id={`${tabsRef}_tab_${tabsIndex}`}
+				aria-controls={`${tabsRef}_panel_${tabsIndex}`}
+				onClick={onClick}
+				className={classNames} {...other}>
+				{children}
+			</li>
+		);
+	}
+});
+
+/**
+ * @module TabsList
+ * Direct child of `Tabs`.
+ * Renders a `nav` and `ul` to contain tab conrols
+ */
+export const TabsList = React.createClass({
+	propTypes: {
+		full: React.PropTypes.bool
+	},
+	defaultProps: {
+		full: false
+	},
+	renderChildren() {
+		const { tabsRef, children } = this.props;
+
+		return React.Children.map(children, (kid, index) => {
+			if (kid.type === TabsListTab) {
```

Comment:
It's better to separate your type checking from the render logic - use a custom function in `propTypes` to check for invalid children so that the React component lifecycle stays consistent:

```jsx
TabsList.propTypes = {
  full: React.PropTypes.bool.isRequired,
  children(props, propName, componentName) {
    const prop = props[propName];
    let error = null;
    React.Children.forEach(prop, child => {
      if (child.type !== TabsListTab) {
        error = new Error(`'${componentName}' children should be of type 'TabsListTab'.`);
      }
    })
    return error;
  }
}
```